src/WebJobs.Script/Extensions/IServiceProviderExtensions.cs

```diff
@@ -21,11 +21,19 @@ public static class IServiceProviderExtensions
                 throw new ArgumentNullException(nameof(serviceProvider));
             }
 
-            var hostManager = serviceProvider.GetService<IScriptHostManager>();
-            if (Utility.TryGetHostService(hostManager, out TService service))
+            try
             {
-                return service;
+                var hostManager = serviceProvider.GetService<IScriptHostManager>();
+                if (Utility.TryGetHostService(hostManager, out TService service))
+                {
+                    return service;
+                }
             }
+            catch
+            {
+                // can get exceptions if the host is being disposed
```

Comment:
Applying the same error handling that Utility.TryGetHostService uses [here](https://github.com/Azure/azure-functions-host/blob/405a25dc064d3cc85f5c6419361db87c610386a1/src/WebJobs.Script/Utility.cs#L62)